dev/ansible/04_test_persistence.yml

```diff
@@ -0,0 +1,192 @@
+  failed_when: nomad_postgres_status_1.stdout == "running"
+  tags: test
+
+- name: Delete terraform related files and folders
+  file:
+    path: "../../example/{{ line_item }}"
+    state: absent
+  loop:
+    - terraform.tfstate.backup
+    - terraform.tfstate
+    - .terraform
+  loop_control:
+    loop_var: line_item
+  tags: test
+
+- name: Terraform apply to restart purged postgres job
+  terraform:
+    project_path: ../../example
+    force_init: true
+    state: present
+  register: terraform
+  tags: test
+
+- name: Terraform stdout
+  debug:
+    msg: "{{terraform.stdout}}"
+  tags: test
+
+- name: Verify that table exist by inserting data into table
+  shell: |
+    psql "dbname=metastore host=127.0.0.1 user={{ username.stdout }} password={{ password.stdout }} port=9999 sslmode=disable" < db_testfiles/data.sql
+  register: insert_data_output
+  failed_when: insert_data_output.stderr != ""
+  tags: test
+
+#****************************************#
+#Test Scenario 2
+#****************************************#
+# 1.Simulate container crash (docker cli, stop container and remove it)
+# 2.Nomad should automatically create a new instance of container with the same volume
+# 3.Verify that data available on new instance
+
+- name: Stop the docker container running postgres
+  shell: docker stop $(docker ps -a -q --filter ancestor="postgres:12-alpine")
+  register: docker_output
+  tags: test
+
+- name: Wait till nomad restarts the docker container running postgres
+  shell: docker ps -a -q --filter ancestor="postgres:12-alpine"
+  register: docker_output1
+  retries: 20
+  delay: 15
+  until: docker_output1.stdout != ""
+  tags: test
+
+- name: Start local proxy on port 9999 to postgres service
+  shell: nohup consul connect proxy -service postgres-local -upstream postgres:9999 -log-level debug </dev/null >/dev/null 2>&1 &
+  when: not lookup('env', 'consul_acl') | bool or (lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'allow')
+  tags: test
+
+- name: Create intention for proxy between postgres-local and posgres
+  shell: consul intention create postgres-local postgres
+  when: lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'deny'
+  tags: test
+
+- name: Start local proxy on port 9999 to postgres service with token
+  shell: nohup consul connect proxy -service postgres-local -upstream postgres:9999 -log-level debug </dev/null >/dev/null 2>&1 &
+  when: lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'deny'
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul_token }}"
+  tags: test
+
+- name: Verify that table exist by inserting data into table
+  shell: |
+    psql "dbname=metastore host=127.0.0.1 user={{ username.stdout }} password={{ password.stdout }} port=9999 sslmode=disable" < db_testfiles/data.sql
+  register: insert_data_output2
+  failed_when: insert_data_output2.stderr != ""
+  tags: test
```

Comment:
CD/CI could fail here due to that proxy has not loaded yet. U might need to add `retries` here and 
`until`

Reply:
That's a good point - will do that!